### ISSUE TITLE
Missing syntax for MonadReader and MonadState

### ISSUE THREAD

**Issue by kcsongor** [scalaz/scalaz#1235]
... and many other type classes.

Also, some syntax file names end in Syntax and others in Ops, although nothing seems to justify this. This one is just bike-shedding, my actual issue at hand is of course the missing syntax for MonadReader and MonadState. Are there plans for adding these?

**Comment by kcsongor** [comment]
(again, I can PR them, if I receive confirmation that it would potentially be merged)

**Comment by kcsongor** [comment]
I realise now that it doesn't make sense for these classes to have 'syntax'

**Comment by puffnfresh** [comment]
Doesn't it make sense?

**Comment by kcsongor** [comment]
Does it? In my understanding, `syntax` provides object.method(arg) style calls, but for the MonadReader functions, you don't really have an object to begin with (only perhaps the implicit type class instance), as opposed to, say, `map`ping on a Functor, or `>>=`ing on a Monad.
If I misunderstand the purpose of syntax, it might make sense, but it won't give me what I originally wanted, that is to have `ask` in the scope of my for comprehensions, which I guess I can get by mixing the MonadReader trait in my class.

Also, the 'createSyntax' flag is set to false explicitly for these type classes in the build system

**Comment by puffnfresh** [comment]
Monoid's `mzero` is a value provided via syntax:

https://github.com/scalaz/scalaz/blob/series/7.3.x/core/src/main/scala/scalaz/syntax/MonoidSyntax.scala#L25

Syntax is not just for "enrichment" style methods. We should not be requiring people to extend `MonadReader` into classes just to get access to the methods.

**Comment by kcsongor** [comment]
Good point, I definitely missed that. Do you know why no syntax is being generated for these classes? https://github.com/scalaz/scalaz/blob/9eeccf5f2634c6dd91127ccca3409d7aec4998e3/project/GenTypeClass.scala#L75
